src/main/java/ru/kata/spring/boot_security/demo/configs/service/SecurityUserService.java

```diff
@@ -0,0 +1,41 @@
+package ru.kata.spring.boot_security.demo.configs.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import ru.kata.spring.boot_security.demo.models.Role;
+import ru.kata.spring.boot_security.demo.models.User;
+import ru.kata.spring.boot_security.demo.repositiories.UserRepository;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+@Service
+public class SecurityUserService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Autowired
+    public SecurityUserService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    @Transactional
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        User user = userRepository.findByEmail(email);
+        if (user == null) {
+            throw new UsernameNotFoundException("User not found");
+        }
+        return new org.springframework.security.core.userdetails.User(user.getEmail(), user.getPassword(),
+                mapRolesToAuthorities(user.getRoles()));
+    }
```

Comment:
```suggestion
        return new org.springframework.security.core.userdetails.User(
                                 user.getEmail(), 
                                 user.getPassword(),
                                 mapRolesToAuthorities(user.getRoles())
        );
    }
```
если еще и автовыравнивание прожать, то вообще по красоте будет, так читабельнее ИМХО

Reply:
fixed